flang/test/Lower/OpenMP/critical.f90

```diff
@@ -51,3 +51,27 @@ subroutine predetermined_privatization()
   end do
   !$omp end parallel do
 end
+
+! https://github.com/llvm/llvm-project/issues/75767
+!CHECK-LABEL: func @_QPnested_privatization(
+subroutine nested_privatization()
```

Comment:
Looking again, `nested_privatization` doesn't seem to properly describe the scenario where the issue happened.
Maybe `parallel_critical_privatization` would be a better name?